rasa/nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -1007,11 +1028,10 @@ def _prepare_entity_recognition_layers(self) -> None:
             average="micro",
         )
 
-    def set_training_phase(self, training: bool) -> None:
-        if training:
-            self.training = tf.ones((), tf.bool)
-        else:
-            self.training = tf.zeros((), tf.bool)
+    @staticmethod
+    def _get_mask_and_lengths(mask):
+        sequence_lengths = tf.cast(tf.reduce_sum(mask[:, :, 0], 1), tf.int32)
+        return mask, sequence_lengths
```

Comment:
`mask` is not modified, why does the function return it?

Reply:
because I call it 2 times, once for predict, once for train, so I thought it is less error prone to reduce the call to 1 line

Reply:
we could simply save mask into a local variable before calling the function

Reply:
what do you mean?

Reply:
instead of 
```
        mask_text, sequence_lengths = self._get_mask_and_lengths(
            tf_batch_data["text_mask"][0]
        )
```
use
```
        mask_text = tf_batch_data["text_mask"][0]
        sequence_lengths = self._get_mask_and_lengths(
            mask_text
        )
```